compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -816,6 +816,17 @@ impl<'context> Elaborator<'context> {
             None
         };
 
+        let attributes = func.secondary_attributes().iter();
+        let attributes = attributes.filter_map(|secondary_attribute| {
+            if let SecondaryAttribute::Custom(name) = secondary_attribute {
+                Some(name.clone())
+            } else {
+                None
+            }
+        });
+        let attributes: Vec<_> = attributes.collect();
+        let attributes = if attributes.is_empty() { None } else { Some(attributes) };
```

Comment:
We can store just a `Vec` here, the None variant is the same as an empty vector

Also, can you add a `SecondaryAttribute::as_custom() -> Option<&str>` for line 821?

Reply:
Done!